### PR TITLE
Replace base branch token with new branch token in fork request

### PR DESCRIPTION
**Pull request by norberthu**
Pass new run ID into the new branch token initialization.

## What changed?
Replace ForkBranchToken with NewBranchToken in InternalForkHistoryBranchRequest. The new branch token is derived from the base branch token and contains the updated set of ancestors.

### CODE REVIEW

File: common/persistence/data_interfaces.go [hunk 1]
```diff
@@ -877,6 +877,8 @@ type (
 		ForkNodeID int64
 		// the info for clean up data in background
 		Info string
+		// the new run ID
+		NewRunID string
```
[the remark this answers is not in the document]
Comment (yux0) [reply common/persistence/data_interfaces.go:881]: What is the new run id for conflict resolution case?

File: service/history/ndc/branch_manager.go [hunk 2]
```diff
@@ -223,13 +223,15 @@ func (r *BranchMgrImpl) createNewBranch(
 	executionInfo := r.mutableState.GetExecutionInfo()
 	namespaceID := executionInfo.NamespaceId
 	workflowID := executionInfo.WorkflowId
+	runID := uuid.New()
 
 	resp, err := r.executionMgr.ForkHistoryBranch(ctx, &persistence.ForkHistoryBranchRequest{
 		ForkBranchToken: baseBranchToken,
 		ForkNodeID:      baseBranchLastEventID + 1,
-		Info:            persistence.BuildHistoryGarbageCleanupInfo(namespaceID, workflowID, uuid.New()),
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/ndc/branch_manager.go:230]: yes. It should he the runID.

[the remark this answers is not in the document]
Comment (norberthu) [reply service/history/ndc/branch_manager.go:230]: yes as in it should use the base run id or the new run id?

[the remark this answers is not in the document]
Comment (norberthu) [reply service/history/ndc/branch_manager.go:230]: switching to `r.mutableState.GetExecutionState().RunId`

File: common/persistence/data_interfaces.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (norberthu) [reply common/persistence/data_interfaces.go:881]: it's just the new run id that corresponds to the new branch

**Review by yux0 — APPROVED** [review]

**Pull request merged**